hera_pspec/pspecdata.py

```diff
@@ -431,10 +402,9 @@ def set_C(self, cov):
         self.clear_cov_cache(cov.keys())
         for key in cov: self._C[key] = cov[key]
 
-    def C_empirical(self, key):
+    def C_model(self, key, mtype='empirical'):
```

Comment:
Maybe `model=` instead of `mtype=`?
I'm a bit worried about consistency here by the way. What happens if the user calls this with one model type, and then tries to use another model type later on? Is the cache dropped and recomputed in this case? Or should we store one cache per model type?

Reply:
It's also unclear to me how we will support more sophisticated covariance matrix methods in future. Should we be getting the user to use `set_C` manually? Or should we attach some wrapper object, similar to `PSpecBeam`, that handles all of the covmat stuff?

Reply:
Yeah that's a good point about the cache. As it stands there is only one slot for a covariance matrix, regardless of its type, so the most recently "set" covmat is the one it will pick up on. Perhaps when we re-haul inverse covariance weighting we should take a closer look at how we implement the storing of various covariance matrix types.